app/services/weights_and_measures.rb

```diff
@@ -24,16 +24,25 @@ def system
 
   UNITS = {
     'weight' => {
+      0.001 => { 'name' => 'mg', 'system' => 'metric' },
       1.0 => { 'name' => 'g', 'system' => 'metric' },
+      1000.0 => { 'name' => 'kg', 'system' => 'metric' },
+      1_000_000.0 => { 'name' => 'T', 'system' => 'metric' },
+
+      28.349523125 => { 'name' => 'oz', 'system' => 'imperial' },
       28.35 => { 'name' => 'oz', 'system' => 'imperial' },
+      453.59237 => { 'name' => 'lb', 'system' => 'imperial' },
       453.6 => { 'name' => 'lb', 'system' => 'imperial' },
-      1000.0 => { 'name' => 'kg', 'system' => 'metric' },
-      1_000_000.0 => { 'name' => 'T', 'system' => 'metric' }
     },
     'volume' => {
       0.001 => { 'name' => 'mL', 'system' => 'metric' },
+      0.01 => { 'name' => 'cL', 'system' => 'metric' },
+      0.1 => { 'name' => 'dL', 'system' => 'metric' },
       1.0 => { 'name' => 'L', 'system' => 'metric' },
-      1000.0 => { 'name' => 'kL', 'system' => 'metric' }
+      1000.0 => { 'name' => 'kL', 'system' => 'metric' },
+
+      0.25 => { 'name' => 'cu', 'system' => 'imperial' },
```

Comment:
As you have pointed out, this is actually a metric cup, so should we not label it as such?

```suggestion
      0.25 => { 'name' => 'cu', 'system' => 'metric' },
```

Reply:
This is tough. It's an imperial measure made to fit into the metric system. It's not included in the metric system which only uses multipliers of 10. But it's not an old imperial unit either. It's mainly used in Australia, Canada and New Zealand. It probably sits somewhere in between and belongs to neither, the imperial system or the metric system.

> Although derived from the [metric system](https://en.wikipedia.org/wiki/Metric_system), it is not an [SI unit](https://en.wikipedia.org/wiki/SI_unit).[[7]](https://en.wikipedia.org/wiki/Cup_(unit)#cite_note-7)

https://en.wikipedia.org/wiki/Cup_(unit)

What do you think?

The other question is: does it matter? Now I came across the UnitPrice service which seems to assume that a variant's unit_value is always expressed in grams, litres, pounds or items. I actually find that class super confusing and I'm not sure it's covering all the cases or why we have to convert this way. When a product is created, the user chooses the unit and that should be enough, shouldn't it?

According to that logic, it does look safer to assign the metric system to a cup so that it doesn't assume it to be a weight. It looks like there's some bigger picture clean-up to do.

Reply:
Hmm interesting. It's still commonly labelled "metric cup" which I would consider close enough.

But does it matter? No, in fact I would suggest we don't bother supporting this unit at all. We've clearly come this far without it.
Until it's [clarified](https://github.com/datafoodconsortium/taxonomies/issues/8), or as you suggest we avoid scaling it and just treat it as a label, we probably _shouldn't_ implement it.

Another thought: if "cup" is just a label that doesn't have a standard meaning, then why don't we import it as an `items` label? Could this work?

```ruby
    when quantity_unit.CUP
      # Interpreted as a label, because this is not an internationally measurable unit.
      # https://github.com/datafoodconsortium/taxonomies/issues/8
      ["items", "cup", 1]
```

Reply:
> But does it matter? No, in fact I would suggest we don't bother supporting this unit at all.

Of course, thank you for thinking outside the cup and identifying this issue clearly as storm in a cup.

I also like your suggestion of importing the label. But there are a lot more DFC measures with a label we just import as generic "items". I would like to change that logic in general to use a label for all "items".